include/onnxruntime/core/framework/allocator.h

```diff
@@ -37,6 +37,8 @@ namespace onnxruntime {
 constexpr const char* CPU = "Cpu";
 constexpr const char* CUDA = "Cuda";
 constexpr const char* CUDA_PINNED = "CudaPinned";
+constexpr const char* ROCM = "Rocm";
+constexpr const char* ROCM_PINNED = "RocmPinned";
```

Comment:
Let's not add new type for now. I had a PR to rename CUDA/CUDA_PINNED to GPU/GPU_PINNED and share it between CUDA and ROCm. For some weird failure, the PR couldn't be merged. Whenever I have cycles, I will re-visit that PR again.

Reply:
Fixed by commit 0ae9ef56c4a53f7f664b18780ee192b8d4acf0e5.